### DISCUSSION TITLE
How to find all rows that have a null value for a nullable json field?

### DISCUSSION THREAD

**Discussion by paymog** [prisma/prisma#21216] [Q&A] [answered]
### Question

I have a model like so

```prisma
model BillingSubscription {
  id                      String         @id // this will use the actual stripe subscription ID for easy lookup in either system
  product_id              String
  paid_in_full            Boolean
  period_end              DateTime?      @db.Timestamp(3)
  billing_threshold_cents Int?
  subscription_items      Json?
  full_subscription       Json?
  billingProduct          BillingProduct @relation(fields: [product_id], references: [id], onDelete: Cascade)
  project                 Project[]
}
```

How can I query for all rows that have a null value for `full_subscription`? I've tried the following

```
  const subscriptions = await prisma.billingSubscription.findMany({
    where: {
      full_subscription: null,
    },
  });
```

But this gives me the following error:

```
error TS2322: Type 'null' is not assignable to type 'JsonNullableFilter<"BillingSubscription"> | undefined'.

19       full_subscription: null,
         ~~~~~~~~~~~~~~~~~
```

I see that I can pass `undefined` to the filter like so:

```
  const subscriptions = await prisma.billingSubscription.findMany({
    where: {
      full_subscription: undefined,
    },
  });
```

But it's not clear that will execute the query I want based on [these docs](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined?query=where&page=1).

**Comment by paymog** [comment, marked as answer]
found the answer! https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-by-null-values

**Reply by raphaelbadia** [reply]
Link is dead, here is the new link https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/working-with-json-fields
